tools/functional-tester/etcd-tester/util.go

```diff
@@ -33,3 +38,43 @@ func max(n1, n2 int64) int64 {
 	}
 	return n2
 }
+
+type serverStats struct {
+	State string `json:"state"`
+}
+
+func getNonLeader(endpoints ...string) (int, string) {
+	for i, endpoint := range endpoints {
+		sts := serverStats{}
+		resp, err := http.Get(endpoint + "/v2/stats/self")
+		if err != nil {
+			continue
+		}
+		if err := json.NewDecoder(resp.Body).Decode(&sts); err != nil {
+			continue
+		}
+		resp.Body.Close()
+		if sts.State != "StateLeader" {
+			return i, endpoint
+		}
+	}
+	return 0, ""
+}
+
+func getLeader(endpoints ...string) (int, string) {
+	for i, endpoint := range endpoints {
+		sts := serverStats{}
+		resp, err := http.Get(endpoint + "/v2/stats/self")
```

Comment:
we should not rely on v2 api. Probably we should add a new api into Cluster service in v3 proto to return the current leader.